gemini/use-cases/education/resume_generator.ipynb

```diff
@@ -0,0 +1,828 @@
+        "        if isinstance(block, Table):  # check block is table\n",
+        "            for row in block.rows:  # each table row\n",
+        "                tab_row = [cell.text for cell in row.cells]\n",
+        "                tab_row = list(\n",
+        "                    filter(lambda x: x != \"\", tab_row)\n",
+        "                )  # remove empty strings\n",
+        "                if len(tab_row) > 1 and len(set(tab_row)) == 1:\n",
+        "                    tab_row = tab_row[0]\n",
+        "                    parsed_doc += \"\".join(tab_row) + \"\\n\"\n",
+        "                else:\n",
+        "                    parsed_doc += \", \".join(tab_row) + \"\\n\"\n",
+        "        else:\n",
+        "            parsed_doc += block.text + \"\\n\"  # paragraph text\n",
+        "    return parsed_doc"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "f2541c9f-8d4a-4de5-80af-1d2561ceb3c7"
+      },
+      "outputs": [],
+      "source": [
+        "def parse_resume(res_path: str) -> str:\n",
+        "    \"\"\"\n",
+        "    Driver method to parse resume with fallbacks.\n",
+        "\n",
+        "    Args:\n",
+        "        res_path: Path to the resume file\n",
+        "\n",
+        "    Returns:\n",
+        "        The parsed resume as string\n",
+        "    \"\"\"\n",
+        "    file_type = res_path.rsplit(\".\", 1)[-1]\n",
+        "    if file_type == \"docx\":\n",
+        "        parsed_resume = convert_docx_to_text(file_path=res_path).strip()\n",
+        "        # check if name, address, email address and mobile no. has been extracted\n",
+        "        personal_details = basic_info_extraction(parsed_resume)\n",
+        "        if \"Not Found\" in personal_details.values():\n",
+        "            parsed_resume_2 = docx2txt.process(res_path)\n",
+        "            per_details_2 = \"\"\n",
+        "            personal_details_2 = basic_info_extraction(parsed_resume_2)\n",
+        "            for k, v in personal_details_2.items():\n",
+        "                per_details_2 += f\"{k}: {v}\\n\"\n",
+        "            parsed_resume = per_details_2 + \"\\n\" + parsed_resume\n",
+        "    elif file_type == \"pdf\":\n",
+        "        parsed_resume = read_pdf(res_path).strip()\n",
+        "    elif file_type == \"txt\":\n",
+        "        with open(res_path) as resume_text:\n",
+        "            parsed_resume = resume_text.read().strip()\n",
+        "    else:\n",
+        "        raise Exception(\"File is not of valid type\")\n",
+        "    return parsed_resume"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "7924162c-7318-4c56-a673-a7411e6fdfd8"
+      },
+      "outputs": [],
+      "source": [
+        "def structure_response(resume_generated: str) -> str:\n",
```

Comment:
I think this typing is incorrect. Should probably be a dict.

Reply:
There was an unused dict which might have caused the confusion. I have removed it. The return type should be string because this is the final output which we are printing